### PR TITLE
Update AbstractArchiveTask.isPreserveFileTimestamps to Property

### PR DESCRIPTION
**Pull request by tresat**

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.reproducible-archives.gradle.kts [hunk 1]
```diff
@@ -16,8 +16,26 @@
 
 // Ensure the archives produced are reproducible
 tasks.withType<AbstractArchiveTask>().configureEach {
-    isPreserveFileTimestamps = false
-    isReproducibleFileOrder = true
+    // TODO: Remove these try-catches once https://github.com/gradle/gradle/pull/31212 is merged, the wrapper for the Gradle build updated, and the properties are removed
```
Comment (tresat) [inline build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.reproducible-archives.gradle.kts:19]: 💭 This is a hacky workaround.  IDK if there is a better way to handle usage of properties in the gradle build itself that will be updated.

**Review by asodja — APPROVED** [review]
Did a small polish, but other than that LGTM, thanks!

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test RFN

**Comment by asodja** [comment]
@bot-gradle test RFN

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
> Ah, ok, so that's the preferred way to deal with properties used in our own build. I suppose that's less hacky than guess-and-check with reflection. Thanks.

Your changes were ok. I just forgot to mention, that for boolean properties we add `getIsX` deprecated menthods, just to keep source compatibility

**Comment by asodja** [comment]
@bot-gradle test RFN

**Pull request merged**